# Notes from meeting with Katie 

We discussed in this meeting **steps for prepping LC18 (2018OAExp_larvae) samples for sequencing**

### Step 1 Confirmation of samples
* Determine # of **zygote** and **larvae** samples
* Practice extractions on block 3 zygotes (happening this Friday)
  * Test on Omega Blood and Tissue Kits
  * IF available also try Omega mollusc (expired) and kit jon used (if available)
* How many viable samples are we able to extract?
* Decide best on available samples which samples we will focus on

### Step 2 Barcoding and Sequencing Plan
* Figure out the barcode design
  * option 1 - **single index design** - order 18-24 more oligos for single index (kit comes with 6, see revised plan for more info)
  * option 2 - **dual index design** - order 6 oligos for i5  (**should confirm with zymo this is a good idea**)
* Get GENEWIZ quotes (**Email sent**)
  * Asked for info about NovaSeq S4 which has an advertised cost of 6800$.
    * this cost may enable more samples (up to 78 based on earlier calculations) or >25x coverage at a lower cost than the NovaSeq S2 platform through harvard.
* Hone in on specific questions (this will inform which samples to prioritize)

### Step 3 Library Prep
* Positive controls
  * Mantle tissue from previously analyzed samples (AE17 sample) to confirm library prep works.
    * Not sequenced
  * Methylated DNA control for BS conversion
    * Look into this a bit more
    * One option: zymo D5012-mouse (check there isn't an invertebrate one)
  * Unmethylated DNA control
    * promega D1521
* Negative Control
  * water

### Step 4 PCR and Quantification
* Things to consider
  * Should we be doing a final size selection post PCR if we dont fragment and size select prior to BS conversion?
  * Should we do a QPCR or tape station for each sample?
* Reagents we need to order
  * Qubit HS reagents
  * Agilent Tapes (?)
  * QPCR reagents (?)
  * Filtered Tips

Notes for reaching out to GENEWIZ
* Request quotes for sequencing
* Confirm index (primer) design will work on the sequencing platform we are using

Notes for reaching out to Zymo
* Confirm the primer design they use for the pico methylation kit
  * On there website the mention the smal RNA index / primers but these don't match the information the rep provided to Katie.
* Discuss using custom oligos in a dual index design (to cut down on the number of oligos we are order, which could be important if we can sequence more samples).

  
